src/main/java/cubicchunks/server/ServerCubeCache.java

```diff
@@ -405,6 +419,11 @@ public void unloadCube(int cubeX, int cubeY, int cubeZ) {
 		if (cubeIsNearSpawn(cubeX, cubeY, cubeZ)) {
 			return;
 		}
+		
+		// Do not unload cubes which are required for generating currently loaded cubes.
+		if (dependencyManager.isRequired(cubeX, cubeY, cubeZ)) {
+			return;
```

Comment:
There we go: https://gist.github.com/Barteks2x/f988a465ebdd59e03349
It needs to be ported from M3L mappings to MCP mappings and to the changes in CubeCache classes, but it should be still useful. (iirc) It shows red square with player in the center, each square is a column, with a number representing number of loaded cubes. Loaded columns are blue.